src/drivers/imu/mpu9250/mpu9250.cpp

```diff
@@ -995,11 +1178,51 @@ MPU9250::gyro_ioctl(struct file *filp, int cmd, unsigned long arg)
 	}
 }
 
+int
+MPU9250::select_register_bank(uint8_t bank) {
+	uint8_t ret;
+	//if (bank != _selected_bank ) {
+	uint8_t buf;
+	uint8_t count=0;
+
+
+	ret = _interface->write(MPU9250_LOW_SPEED_OP(ICMREG_20948_BANK_SEL), &bank, 1);
+	if (ret != OK) {
+		return ret;
+	}
+	up_udelay(100);
```

Comment:
This stalls the complete CPU. Don't do this in a driver in general and ONLY as exception during setup. This is critical to resolve.

Reply:
Thanks - replaced up_udelay() with usleep().